src/loaddb/load_server_loader.cpp

```diff
@@ -114,8 +114,171 @@ namespace cubload
   LC_FIND_CLASSNAME
   server_class_installer::locate_class (const char *class_name, OID &class_oid)
   {
+    LC_FIND_CLASSNAME found = LC_CLASSNAME_EXIST;
+
     cubthread::entry &thread_ref = cubthread::get_entry ();
-    return xlocator_find_class_oid (&thread_ref, class_name, &class_oid, BU_LOCK);
+
+    found = xlocator_find_class_oid (&thread_ref, class_name, &class_oid, BU_LOCK);
+    if (found == LC_CLASSNAME_EXIST)
+      {
+	return found;
+      }
+
+    /* This is the case when the loaddb utility is executed with the --no-user-specified-name option as the dba user. */
+    if (thread_ref.conn_entry->client_type == DB_CLIENT_TYPE_ADMIN_UTILITY
+	&& prm_get_bool_value (PRM_ID_NO_USER_SPECIFIED_NAME))
+      {
+	cubthread::entry &thread_ref = cubthread::get_entry ();
+	LC_FIND_CLASSNAME found_again = LC_CLASSNAME_EXIST;
+	HEAP_CACHE_ATTRINFO attr_info;
+	int attr_idx = 0;
+	HEAP_SCANCACHE scan_cache;
+	SCAN_CODE scan_code = S_SUCCESS;
+	RECDES recdes = RECDES_INITIALIZER;
+	HFID hfid = HFID_INITIALIZER;
+	OID inst_oid = OID_INITIALIZER;
+	HEAP_ATTRVALUE *heap_value = NULL;
+	int error = NO_ERROR;
+	int i = 0;
+
+	error = heap_attrinfo_start (&thread_ref, oid_User_class_oid, -1, NULL, &attr_info);
+	assert (attr_info.num_values != -1);
+	if (error != NO_ERROR)
+	  {
+	    ASSERT_ERROR ();
+	    return LC_CLASSNAME_ERROR;
+	  }
+
+	error = heap_scancache_quick_start_root_hfid (&thread_ref, &scan_cache);
+	if (error != NO_ERROR)
+	  {
+	    ASSERT_ERROR ();
+	    heap_attrinfo_end (&thread_ref, &attr_info);
+	    return LC_CLASSNAME_ERROR;
+	  }
+
+	scan_code = heap_get_class_record (&thread_ref, oid_User_class_oid, &recdes, &scan_cache, PEEK);
+	if (scan_code != S_SUCCESS)
+	  {
+	    ASSERT_ERROR ();
+	    heap_scancache_end (&thread_ref, &scan_cache);
+	    heap_attrinfo_end (&thread_ref, &attr_info);
+	    return LC_CLASSNAME_ERROR;
+	  }
+
+	{
+	  char *string = NULL;
+	  int alloced_string = 0;
+
+	  for (i = 0; i < attr_info.num_values; i++)
+	    {
+	      error = or_get_attrname (&recdes, i, &string, &alloced_string);
+	      if (error != NO_ERROR)
+		{
+		  ASSERT_ERROR ();
+		  heap_attrinfo_end (&thread_ref, &attr_info);
+		  break;
+		}
+
+	      if (string != NULL && strcmp ("name", string) == 0)
+		{
+		  attr_idx = i;
```

Comment:
The code is written on the assumption that "name" must be present among att_names.
Is that right? Otherwise, there will be a problem on line 242.

Reply:
The db_user system class has a "name" column.
So there should never be a case where attr_idx is not set.

But even if it is not set, the code below can be executed, so does the defense code need to be added?

Reply:
If it's guaranteed, there's no need to fix it.
But, If it is a fixed system table, wouldn't it be possible to define idx as a constant value without looking for "name"?

Reply:
ATTR_IDX is fixed to 7 unless the column order in the db_user table is changed or a new column is added after the "name" column.

I will improve it by adding the macro variable below.
```
#define CATCLS_USER_ATTR_IDX_NAME 7
```

And if a change occurs in the code for adding a column to the db_user system table, I will leave a comment requesting to change CATCLS_USER_ATTR_IDX_NAME together.